Coding cheat sheet from docs-code-eval repo

mdlinville opened the pull request:
## Description

1. Adds a script to :
    - Sync ground-truth code examples from docs-code-eval repo into snippets in this repo
    - Include all snippets in logical order in models/ref/sdk-coding-cheat-sheet.mdx
1. Use script to generate the cheat sheet and initial snippets.
1. Add an on-demand Github action to regenerate the snippets and cheat sheet on demand, generating a drsft PR if updates are available

## Testing
- [x] Used the script to generate the initial contents
- [x] Local build succeeds without errors (`mint dev`)
- [x] Local link check succeeds without errors (`mint broken-links`)
- [x] PR tests succeed

mdlinville committed:
Add SDK coding cheat sheet synced from docs-code-eval

- Add sync_code_examples.sh script to pull ground truth examples
- Add generate_cheat_sheet.py to create organized cheat sheet page
- Copy 30 code examples from docs-code-eval/ground_truth
- Generate comprehensive SDK cheat sheet at models/ref/sdk-coding-cheat-sheet.mdx
- Organize examples by category (Artifacts, Registry, Logging, Run Management)
- Reuses evaluation benchmark ground truth as developer resource
- Add README for code examples directory

Refactor cheat sheet into multi-page structure with landing page

Remove duplicate .py snippets, clean up logic

Remove Additional Resources sections

Update README to fix a link and adhere to docs style

mdlinville committed:
Add GitHub Action for automated code examples sync

- Add workflow to sync ground truth examples from docs-code-eval
- Runs on-demand (workflow_dispatch)
- Creates draft PR only if changes detected
- Includes change summary and review checklist
- Auto-labels PRs: documentation, automated, code-examples
- Workflow documentation in .github/workflows/README.md

ngrayluna commented on snippets/code-examples/artifact_add_alias.py:
```diff
@@ -0,0 +1,13 @@
+"""
+Add one or more aliases to an artifact when logging it to W&B.
```
Do we need both `snippets/codes-examples` and `en/_includes/code-examples` directories?

mdlinville replied:
Done

ngrayluna commented on models/ref/sdk-coding-cheat-sheet/artifacts.mdx:
```diff
@@ -0,0 +1,159 @@
+---
+
+
+## Additional Resources
```
This section seems out of place and incomplete. Let's remove it.

mdlinville replied:
Done

johndmulhausen commented on snippets/code-examples/README.md:
```diff
@@ -0,0 +1,80 @@
+# W&B SDK code examples
```
Probably shouldn't check this in -- add to .gitignore?

mdlinville replied:
No, this is a new README for `wandb/docs` that documents that these files are generated and from where.

johndmulhausen commented on snippets/code-examples/llm_evaluation_tasks.csv:
```diff
@@ -0,0 +1,32 @@
+Category,Task Description,Difficulty,Reference Script
```
Probably shouldn't check this in -- add to .gitignore?

mdlinville committed:
Refactors

- Stop using or saving the CSV file, get descriptions from
  the first sentence in the docstring
- Improve snippet inclusion: Save the snippets as .py files
  and create a new <CodeSnippet/> component that accepts a path

mdlinville committed:
Improvements

- Deal with W&B on first mentions
- Consolidate headings
- Deal with mismatched indefinite articles

mdlinville committed:
Fix parsing errors in CodeSnippet component

mdlinville committed:
Go back to using mdx wrappers

mdlinville committed:
Fix component and import statements

mdlinville committed:
Fix import paths

mdlinville committed:
GO back to MDX snippets, remove manual breadcrumbs, remove horizontal rules

mdlinville committed:
Fix warning

mdlinville committed:
Fix more warnings
